From 60d7bb3d7a6a78cab7ef761ec6f4d7479b3689b8 Mon Sep 17 00:00:00 2001
From: John Cabaj <john.cabaj@canonical.com>
Date: Thu, 22 Jan 2026 14:55:06 -0600
Subject: [PATCH] Linux 6.19: handle --werror with CONFIG_OBJTOOL_WERROR=y

Linux upstream commit 56754f0f46f6: "objtool: Rename
--Werror to --werror" did just that, so we should check for
either "--Werror" or "--werror", else the build will fail

Signed-off-by: John Cabaj <john.cabaj@canonical.com>
---
 scripts/objtool-wrapper.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/scripts/objtool-wrapper.in b/scripts/objtool-wrapper.in
index 0451f8718233..0cea6e2f655c 100644
--- a/scripts/objtool-wrapper.in
+++ b/scripts/objtool-wrapper.in
@@ -22,10 +22,10 @@
 # FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 # IN THE SOFTWARE.
 
-# Filter out objtools '--Werror' flag.
+# Filter out objtools '--Werror or --werror' flag.
 
 objtool="@abs_objtool_binary@"
-args=$(echo "$*" | sed s/--Werror//)
+args=$(echo "$*" | sed 's/--Werror\|--werror//')
 
 if [ -z "$objtool" ]; then
 	echo "$(basename "$0"): No objtool binary configured" 1>&2
